### PR TITLE
add menu example to align with figma

### LINKED ISSUE #2358: Menu - Update the 'Variants' section on site
Based on #1776, 

The "[Variants](https://design-system.hpe.design/components/menu?q=me#variants)" section on the Design System site under 'Menu' has to include two visual examples: (a)Menu with a toolbar button and (b)Menu with default button. 

Here's the [Figma file](https://www.figma.com/file/DyFWlzLBUtK5KH5jJhwoCB/HPE-Menu-Component?node-id=2292%3A1223) for reference.

### PR DESCRIPTION
**Pull request by britt6612**
[Deploy Preview](https://deploy-preview-2408--keen-mayer-a86c8b.netlify.app/components/menu#within-a-header-using-a-default-button)

#### What does this PR do?
Adds the menu with `toolbar` button variant 
#### Where should the reviewer start?
menu.mdx
#### What testing has been done on this PR?
site
In addition to the feature you are implementing, have you checked the following:

**General UX Checks**
- [ ] Small, medium, and large screen sizes
- [ ] Cross-browsers (FireFox, Chrome, and Safari)
- [ ] Light & dark modes
- [ ] All hyperlinks route properly

**Accessibility Checks**
- [ ] Keyboard interactions
- [ ] Screen reader experience
- [ ] Run WAVE accessibility plugin (Chrome)

**Code Quality Checks**
- [ ] Console is free of warnings and errors
- [ ] Passes E2E commit checks
- [ ] Visual snapshots are reasonable

#### How should this be manually tested?
site 
#### Any background context you want to provide?
There were 2 variants that were not added to the menu page
One of the variant was already there so I added to the content 
#### What are the relevant issues?
closes #2358 
#### Should this PR be mentioned in Design System updates?
no
#### Is this change backwards compatible or is it a breaking change?
backwards compatible

### CODE REVIEW

File: aries-site/src/examples/components/menu/MenuToolBarExample.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import { Menu } from 'grommet';
+import {
+  FilterControls,
+  FiltersProvider,
+} from '../../templates/FilterControls';
+
+export const MenuToolBarExample = () => (
+  <FiltersProvider>
```
Comment (taysea) [inline aries-site/src/examples/components/menu/MenuToolBarExample.js:9]: We already have an equivalent example on https://design-system.hpe.design/components/button?q=button#toolbar-button -- can we just point to that exmaple on the MDX page?

File: aries-site/src/pages/components/menu.mdx [hunk 2]
```diff
@@ -58,6 +60,19 @@ A menu can be used in a [Header](/components/header) to group related actions.
   <MenuHeaderExample />
 </Example>
 
+### Menu with Toolbar Button
+
+Use the toolbar button variant when it has to be placed in row with other toolbar controls like search or filter.
```
Comment (taysea) [inline aries-site/src/pages/components/menu.mdx:65]: Can we add the sentence from https://design-system.hpe.design/components/button?q=button#toolbar-button that talks about the desire to give the buttons and input equal prominence?

Comment (britt6612) [reply aries-site/src/pages/components/menu.mdx:65]: yes I can add that

File: aries-site/src/examples/components/menu/MenuToolBarExample.js (hunk 1 above)
Comment (britt6612) [reply aries-site/src/examples/components/menu/MenuToolBarExample.js:9]: good point one less file to have

File: aries-site/src/pages/components/menu.mdx [hunk 3]
```diff
@@ -58,6 +60,20 @@ A menu can be used in a [Header](/components/header) to group related actions.
...
 </Example>
 
+### Menu with Toolbar Button
+
+Use the toolbar button variant when it has to be placed in row with other toolbar controls like search or filter. 
+In this situation, a search input, filter button, and action menu are presented as a unified set of controls. The intent is to present each control with equal prominence.
+
+<Example
+  docs="https://v2.grommet.io/menu?theme=hpe#props"
+  code="https://raw.githubusercontent.com/grommet/hpe-design-system/master/aries-site/src/examples/components/menu/MenuToolBarExample.js"
```
Comment (taysea) [inline aries-site/src/pages/components/menu.mdx:70]: Update code link to match

File: aries-site/src/pages/components/menu.mdx [hunk 4]
```diff
@@ -45,9 +46,10 @@ An icon can be used on a menu item to provide additional context about the actio
 
 There are various places a Menu may be useful within an application. If the user is going to be selecting from a list of options to execute an action, Menu is a good component to use. The usage of labels and icons allow a menu to be versatile for different layout contexts.
 
-### Within a header
+### Within a header using a default Button
 
 A menu can be used in a [Header](/components/header) to group related actions.
+Use the default button variant when the menu has to be is placed in a row with other default buttons or with a heading.
```
Comment (taysea) [inline aries-site/src/pages/components/menu.mdx:52]: Wondering if we should be more prescriptive and say this instead:

The default Menu should be used in all instances unless the Menu is contained within a toolbar, in which case see [Menu within a toolbar](#menu-within-toolbar) guidance.

File: aries-site/src/pages/components/menu.mdx [hunk 5]
```diff
@@ -45,9 +46,10 @@ An icon can be used on a menu item to provide additional context about the actio
 
 There are various places a Menu may be useful within an application. If the user is going to be selecting from a list of options to execute an action, Menu is a good component to use. The usage of labels and icons allow a menu to be versatile for different layout contexts.
 
-### Within a header
+### Within a header using a default Button
```
Comment (taysea) [inline aries-site/src/pages/components/menu.mdx:49]:
```suggestion
### Within a header
```
This is just the default menu -- I feel like specifying "using a default Button" makes it seem like the user has to do something extra.

File: aries-site/src/pages/components/menu.mdx [hunk 6]
```diff
@@ -58,6 +60,27 @@ A menu can be used in a [Header](/components/header) to group related actions.
   <MenuHeaderExample />
 </Example>
 
+### Menu with Toolbar Button
```
Comment (taysea) [inline aries-site/src/pages/components/menu.mdx:63]:
```suggestion
### Within a Toolbar
```
Matching the wording of the other section.

File: aries-site/src/pages/components/menu.mdx [hunk 7]
```diff
@@ -58,6 +60,27 @@ A menu can be used in a [Header](/components/header) to group related actions.
...
+
+Use the toolbar button variant when it has to be placed in row with other toolbar controls like search or filter.
```
Comment (taysea) [inline aries-site/src/pages/components/menu.mdx:65]:
```suggestion
Use `kind="toolbar"` when the Menu is placed in row with other toolbar controls like search or filter. 
```

File: aries-site/src/pages/components/menu.mdx (hunk 5 above)
Comment (britt6612) [reply aries-site/src/pages/components/menu.mdx:49]: thats fine I wasnt sure to leave it or try and combine what we have with what Keeth put in Figma

File: aries-site/src/pages/components/menu.mdx (hunk 4 above)
Comment (britt6612) [reply aries-site/src/pages/components/menu.mdx]: good idea

**Review by halocline — APPROVED** [review]
LGTM.

**Review by taysea — APPROVED** [review]

**Pull request merged**